src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts

```diff
@@ -44,6 +42,25 @@ export interface SqlArgs {
 	provider?: string;
 }
 
+//#region decorators
+
+type PathHandler = (uri: URI) => Promise<boolean>;
+
+const pathMappings: { [key: string]: PathHandler } = {};
+
+interface PathHandlerOptions {
+	path: string
+}
+
+function pathHandler({ path }: PathHandlerOptions) {
+	return function (target: any, propertyKey: string, descriptor: PropertyDescriptor) {
+		const method: PathHandler = descriptor.value;
```

Comment:
What happens here if the return type of the method isn't actually a `Promise<boolean>`? It should error out if that isn't the case so we're sure this is handled correctly.

Reply:
I think I would have to use reflect-metadata to do stricter checking of the signature of the method.

 I'm not sure if it's worth, but if you think there is an easier way I would definitely do it. Or if reflect-metadata wouldn't be difficult.

Reply:
Yeah I don't know how you'd do that when decorated. Probably not worth the effort since I imagine this is going to be a rarely used decorator.

Reply:
You could do something like check the returned value and throw if it wasn't the type we wanted - but that requires actually running it which isn't great.

Reply:
couldn't you just await the value returned from the method and check if it is a boolean? That at least enforces it at runtime.